CMakeLists.txt

```diff
@@ -19,16 +19,31 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH}
                       ${CMAKE_SOURCE_DIR}/cmake/macros)
 
 include(Options)
+# If we were given a python executable, we'll extract the name from that to use in
+# the python shebang replacement. This will make these files more portable by not
+# having some full path to a python executable which may not exist on another machine.
+if (DEFINED PYTHON_EXECUTABLE)
+    get_filename_component(PYTHON_EXE_BASENAME ${PYTHON_EXECUTABLE} NAME)
+endif()
+
 include(ProjectDefaults)
 include(Packages)
 
 # This has to be defined after Packages is included, because it relies on the
 # discovered path to the python executable.
-set(PXR_PYTHON_SHEBANG "${PYTHON_EXECUTABLE}" 
-    CACHE 
-    STRING
-    "Replacement path for Python #! line."
-)
+if (DEFINED PYTHON_EXE_BASENAME)
+    set(PXR_PYTHON_SHEBANG "${PYTHON_EXE_BASENAME}" 
+        CACHE 
+        STRING
+        "Replacement path for Python #! line."
+    )
+else()
+    set(PXR_PYTHON_SHEBANG "${PYTHON_EXECUTABLE}" 
+        CACHE 
+        STRING
+        "Replacement path for Python #! line."
+    )
```

Comment:
I think there's another PR targetting this same issue a different way https://github.com/PixarAnimationStudios/USD/pull/1445

Reply:
I looked at that change. It is good if you are using the system python (either 2 or 3). In my changes, I allow passing in the python to use and then extract the py shebang from that.

Reply:
I do think the explicit specification of the shebang as in #1445 is a bit clearer than having special logic that does this in certain cases, i.e. if PYTHON_EXECUTABLE is supplied. 

If we merged that PR (or something like it) in, would it be sufficient for your use cases? I'd imagine since you're supplying the Python executable directly to the build script, you could also specify PXR_PYTHON_SHEBANG with the executable name when you run the build script?

Reply:
Yes that is actually better. I'll remove the changes to this file.